internal/testrunner/runners/system/runner.go

```diff
@@ -182,6 +219,45 @@ func (r *runner) run() (results []testrunner.TestResult, err error) {
 			}
 		}
 	}
+
+	// check agent logs
```

Comment:
Nit. Consider moving this to another function. This may also help to add tests for this.

Reply:
Done. I've added tests for this new function. That made me realize that the ordering of logs was not respected when they were shown.